### PR TITLE
watchyourlan: init at 2.0.4

### PR DESCRIPTION
**Pull request by iv-nn**
## Description of changes

Add WatchYourLan, a lightweight network IP scanner with web GUI.

https://github.com/aceberg/WatchYourLAN

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  vendorHash = "sha256-+7CMubbCrl+DsGmN9/2jCQ6zLDKvjJ6PdJx8iI1vKOQ=";
+
+  src = fetchFromGitHub {
+    owner = "aceberg";
+    repo = "WatchYourLAN";
+    rev = version;
+    hash = "sha256-Z4UitJKWZXGypNJ737DQNGuyDXIbyT7a1wOSc6N2s1k=";
+
+    meta = {
+      description = "Lightweight network IP scanner. Can be used to notify about new hosts and monitor host online/offline history.";
```
Comment (eclairevoyant) [inline pkgs/by-name/wa/watchyourlan/package.nix:18]:
```suggestion
      description = "Lightweight network IP scanner";
      longDescription = ''
        Lightweight network IP scanner. Can be used to notify about new hosts and monitor host online/offline history.
      '';
```

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ pkgs, lib, ... }:
+let
+  inherit (pkgs) buildGoModule fetchFromGitHub;
+in
```
Comment (eclairevoyant) [inline pkgs/by-name/wa/watchyourlan/package.nix:4]:
```suggestion
{ buildGoModule, fetchFromGitHub, lib }:
```

Never pass `pkgs` or use `...` in a package expression

**Review by eclairevoyant — DISMISSED** [review]

File: pkgs/by-name/wa/watchyourlan/package.nix (hunk 1 above)
Comment (iv-nn) [reply pkgs/by-name/wa/watchyourlan/package.nix]: Thanks, I've changed it to "Lightweight network IP scanner with web GUI" which is shorter, only one sentence, while still more informative, as it's not a cli scanner.

**Comment by iv-nn** [comment]
I don't understand `check-maintainers-sorted` error:
```
maintainer ivarmedi (line 9072) should be listed after ivan-tkatchev, not after iv-nn (line 9017)
```
To me ivarmedi *is* after ivan-tkatchev.

**Comment by eclairevoyant** [comment]
I know, the CI error is awful. What it really means is your own entry needs to go elsewhere - though tbh I don't know where offhand.

**Comment by iv-nn** [comment]
OK, it seems `-`and `_` are removed before sorting.

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "aceberg";
+    repo = "WatchYourLAN";
+    rev = version;
+    hash = "sha256-Z4UitJKWZXGypNJ737DQNGuyDXIbyT7a1wOSc6N2s1k=";
+
+    meta = {
+      description = "Lightweight network IP scanner with web GUI";
+      homepage = "https://github.com/aceberg/WatchYourLAN";
+      licence = lib.licenses.mit;
```
Comment (eclairevoyant) [inline pkgs/by-name/wa/watchyourlan/package.nix:21]:
```suggestion
      license = lib.licenses.mit;
```

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    repo = "WatchYourLAN";
+    rev = version;
+    hash = "sha256-Z4UitJKWZXGypNJ737DQNGuyDXIbyT7a1wOSc6N2s1k=";
+
+    meta = {
+      description = "Lightweight network IP scanner with web GUI";
+      homepage = "https://github.com/aceberg/WatchYourLAN";
+      license = lib.licenses.mit;
+      maintainers = [ lib.maintainers.iv-nn ];
+    };
```
Comment (eclairevoyant) [inline pkgs/by-name/wa/watchyourlan/package.nix:23]:
```suggestion
      mainProgram = "WatchYourLAN";
    };
```

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  version = "2.0.0";
+
+  vendorHash = "sha256-+7CMubbCrl+DsGmN9/2jCQ6zLDKvjJ6PdJx8iI1vKOQ=";
+
+  src = fetchFromGitHub {
+    owner = "aceberg";
+    repo = "WatchYourLAN";
+    rev = version;
+    hash = "sha256-Z4UitJKWZXGypNJ737DQNGuyDXIbyT7a1wOSc6N2s1k=";
+
```
Comment (eclairevoyant) [inline pkgs/by-name/wa/watchyourlan/package.nix:17]: Does build, but fails at runtime due to `arp-scan` not being found.

```suggestion

  nativeBuildInputs = [ makeBinaryWrapper ];
  
  postFixup = ''
    wrapProgram $out/bin/WatchYourLAN \
      --prefix PATH : '${lib.makeBinPath [ arp-scan ]}'
  '';
```

Comment (eclairevoyant) [inline pkgs/by-name/wa/watchyourlan/package.nix:17]:
```suggestion
  };
```

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    meta = {
+      description = "Lightweight network IP scanner with web GUI";
+      homepage = "https://github.com/aceberg/WatchYourLAN";
+      license = lib.licenses.mit;
+      maintainers = [ lib.maintainers.iv-nn ];
+    };
+  };
```
Comment (eclairevoyant) [inline pkgs/by-name/wa/watchyourlan/package.nix:24]:
```suggestion
```

**Review by eclairevoyant — DISMISSED** [review]
Even after the fixes below, it seems to fail with a couple of errors:

```
2024/09/01 17:06:13 ERROR mkdir /data: permission denied
2024/09/01 17:06:13 ERROR open /data/WatchYourLAN/config_v2.yaml: no such file or directory
2024/09/01 17:06:13 ERROR open /data/WatchYourLAN/config_v2.yaml: no such file or directory
```

and

```
2024/09/01 17:06:13 INFO Log level=INFO
2024/09/01 17:06:13 INFO Using DB type=sqlite
2024/09/01 17:06:13 ERROR unable to open database file: out of memory (14)
```

**Comment by iv-nn** [comment]
Thanks, I missed a lot of things.

The missing data dir is fixed by using the option `-d /existing/dir`. I've started to write a module to make running this easier.

**Comment by iv-nn** [comment]
@eclairevoyant Are there still requested changes?

**Comment by iv-nn** [comment]
Ofborg test on aarch64-darwin has been on queue for a week, I know the queue is supposed to be long but is this normal? Should I keep waiting? I wanted to update the PR to v2.0.2 but I'm afraid it'll push it back on the queue.

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    wrapProgram $out/bin/WatchYourLAN \
+      --prefix PATH : '${lib.makeBinPath [ arp-scan ]}'
+  '';
+
+  src = fetchFromGitHub {
+    owner = "aceberg";
+    repo = "WatchYourLAN";
+    rev = version;
+    hash = "sha256-nJYDGCkT8vCkxySLONG3OkWVkaBqXqSFgd7N1TTMAf4=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/wa/watchyourlan/package.nix:31]:
```suggestion
```

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  buildGoModule,
+  fetchFromGitHub,
+  makeBinaryWrapper,
+  lib,
+  arp-scan,
+}:
+buildGoModule rec {
+  pname = "watchyourlan";
+  version = "2.0.4";
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/wa/watchyourlan/package.nix:11]:
```suggestion

  src = fetchFromGitHub {
    owner = "aceberg";
    repo = "WatchYourLAN";
    rev = version;
    hash = "sha256-nJYDGCkT8vCkxySLONG3OkWVkaBqXqSFgd7N1TTMAf4=";
  };

```
we usually order things like this

Comment (iv-nn) [reply pkgs/by-name/wa/watchyourlan/package.nix:11]: Done.

File: pkgs/by-name/wa/watchyourlan/package.nix [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  ldflags = [
+    "-s"
+    "-w"
+  ];
+
+  nativeBuildInputs = [ makeBinaryWrapper ];
+
+  postFixup = ''
+    wrapProgram $out/bin/WatchYourLAN \
+      --prefix PATH : '${lib.makeBinPath [ arp-scan ]}'
```
[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/by-name/wa/watchyourlan/package.nix:30]: Not necessary

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**